### PR TITLE
fix: charge code validation rule to validate charge code instead of charge type

### PR DESCRIPTION
**Pull request by MadsDue**
# Description
The validation error currently triggers if you request wholesale subscriptions. Because the charge type "subscription" is larger than 10 chars. This rule should not validate the type, but the charge code instead (cim = mRID)

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)
- https://github.com/Energinet-DataHub/dh3-environments/actions/runs/9805089277

### CODE REVIEW

File: source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/ChargeTypeValidationRule.cs [hunk 1]
```diff
@@ -14,20 +14,20 @@
 
 namespace Energinet.DataHub.Wholesale.Edi.Validation.WholesaleServicesRequest.Rules;
 
-public class ChargeTypeValidationRule : IValidationRule<DataHub.Edi.Requests.WholesaleServicesRequest>
+public class ChargeCodeValidationRule : IValidationRule<DataHub.Edi.Requests.WholesaleServicesRequest>
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/ChargeTypeValidationRule.cs:17]: Did you rename the class along with this?

File: source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/ChargeTypeValidationRule.cs [hunk 2]
```diff
@@ -14,20 +14,20 @@
...
 {
-    private static readonly ValidationError _chargeTypeIdIsToLongError = new ValidationError(
+    private static readonly ValidationError _chargeTypeIdIsToLongError = new(
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/ChargeTypeValidationRule.cs:19]:
```suggestion
    private static readonly ValidationError _chargeCodesLengthInvalid = new(
```

File: source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/ChargeTypeValidationRule.cs [hunk 3]
```diff
@@ -14,20 +14,20 @@
...
         "Følgende chargeType mRID er for lang: {PropertyName}. Den må højst indeholde 10 karaktere/"
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/ChargeTypeValidationRule.cs:20]: Update message to reflect the change from `ChargeType` to `ChargeCode`

File: source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/ChargeTypeValidationRule.cs (hunk 1 above)
Comment (lasrinnil) [inline source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/ChargeTypeValidationRule.cs:17]: Are no rules enforced for the ChargeType then? Only for the ChargeType code?

File: source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/ChargeTypeValidationRule.cs (hunk 3 above)
Comment (MadsDue) [reply source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/ChargeTypeValidationRule.cs:20]: The message is correct :)

File: source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/ChargeTypeValidationRule.cs (hunk 1 above)
Comment (MadsDue) [reply source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/ChargeTypeValidationRule.cs:17]: The chargeType is as I remember enforced by the schema 
subscription=D01, fee=D02 and tariff=D03

**Review by lasrinnil — APPROVED** [review]

**Pull request merged**